src/dmd/dtemplate.d

```diff
@@ -6067,6 +6067,55 @@ extern (C++) class TemplateInstance : ScopeDsymbol
         return hash;
     }
 
+    /*
+        The idea of this function is to see if the template instantiation
+        can be 100% replaced with its eponymous member. All other members
+        can be discarded, even in the compiler to free memory, and can be
+        elided entirely from the binary.
```

Comment:
I might actually make it private since this is all an internal implementation hack to aid optimization rather than a defined api..... just my later goal might need the function to be called somewhere else (which is why it is a separate function to begin with instead of being inline in the codegen function) and I don't know where that is. might be in another module, idk, if I knew how to do it i would have already done it lol